### DISCUSSION TITLE
How to use multiple storage adapters ?

### DISCUSSION THREAD

**Discussion by azevedonik** [directus/directus#8878] [Archive]
Hi ! I'm setting my envs like this:

STORAGE_LOCATIONS="digitalocean, local"

STORAGE_LOCAL_DRIVER="local"
STORAGE_LOCAL_ROOT="./uploads"

STORAGE_DIGITALOCEAN_DRIVER="s3"
STORAGE_DIGITALOCEAN_ROOT="./cms"
STORAGE_DIGITALOCEAN_KEY=
STORAGE_DIGITALOCEAN_SECRET=
STORAGE_DIGITALOCEAN_BUCKET=
STORAGE_DIGITALOCEAN_ENDPOINT="nyc3.digitaloceanspaces.com"

How can I use multiple multiple adapters ? I'm creating two folders local and digitalocean. Wouldn't it send files uploaded to local to the local storage and files uploaded to the digitalocean folder to digitalocean? 

I'd like to have te following

/static --> ./uploads
/digitalocean -> Digitalocean Space

Can you please help me on how to use multiple adapters at once? 
Thanks

**Comment by joselcvarela** [comment]
Hello @azevedonik 

One thing I noticed is we are not trimming values in `STORAGE_LOCATIONS`, so my first advice is to remove whitespaces.
But, at least, the first declaration should have worked.

Check this issue https://github.com/directus/directus/issues/7114
The author had faced same issue as you but it was solved by adding these env variables:
```
STORAGE_DIGITALOCEAN_ROOT="./storage"
STORAGE_DIGITALOCEAN_PUBLIC_URL="/storage"
```

**Reply by azevedonik** [reply]
Hey thanks ! But even removing the white space it doesn't work. It goes to the first adapter. Is it possible to do ? I mean, is it possible to have multiple adapter at the same time and achieve what I mentioned ?

**Reply by joselcvarela** [reply]
It seems on the app, only the first storage is used.
Although, you can defined multiple storage adapters.
This could be useful if you use the API and pretend to upload to one instance or another.
https://docs.directus.io/guides/api-config/#multiple-storage-adapters (check last paragraph of that section)

That being said, you can try to intercept all the uploads and make another request to files by declaring the storage you want use.
https://docs.directus.io/reference/api/system/files/#the-file-object

Or you can use an external mechanism to sync local folder with remote one, like rsync or something.
